mkdocs_macros/errors.py

```diff
@@ -0,0 +1,47 @@
+import textwrap
+import traceback
+from functools import singledispatch
+
+from jinja2 import TemplateSyntaxError
+from mkdocs.structure.pages import Page
+
+
+@singledispatch
+def format_error(error: Exception, markdown: str, page: Page) -> str:
+    """Default error message for a generic exception."""
+    error_type = type(error).__name__
+    return textwrap.dedent(
+        f'''
+        # _Macro Rendering Error_
+
+        _File_: `{page.file.src_path}`
+        
+        _{error_type}_: {error}
+        
+        ```
+        %s
+        ```
+        ''',
+    ).strip() % traceback.format_exc()
+
+
+@format_error.register
```

Comment:
Shouldn't it be `@format_error.register(TemplateSyntaxError)` ?

Reply:
Not necessarily. As per [functools.singledispatch docs](https://docs.python.org/3/library/functools.html#functools.singledispatch),

> For functions annotated with types, the decorator will infer the type of the first argument automatically